// @umijs/bundler-vite/dist/config/transformer/css.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/config/transformer/css.ts
var css_exports = {};
__export(css_exports, {
  default: () => css_default,
  getBrowserlist: () => getBrowserlist
});
module.exports = __toCommonJS(css_exports);
function getBrowserlist(targets) {
  return typeof targets.browsers === "string" ? targets.browser : Object.keys(targets).map((key) => `${key} >= ${targets[key] === true ? "0" : targets[key]}`);
}
var css_default = function css(userConfig) {
  var _a, _b, _c, _d;
  const config = {
    css: { postcss: {}, preprocessorOptions: {} }
  };
  config.css.postcss = __spreadProps(__spreadValues({}, ((_a = userConfig.postcssLoader) == null ? void 0 : _a.postcssOptions) || {}), {
    plugins: [
      ...((_c = (_b = userConfig.postcssLoader) == null ? void 0 : _b.postcssOptions) == null ? void 0 : _c.plugins) || [],
      require("postcss-preset-env")({
        browsers: getBrowserlist(userConfig.targets || {}),
        autoprefixer: __spreadValues({
          flexbox: "no-2009"
        }, userConfig.autoprefixer || {}),
        stage: 3
      }),
      ...userConfig.extraPostCSSPlugins || []
    ]
  });
  config.css.preprocessorOptions.less = __spreadProps(__spreadValues({
    javascriptEnabled: true
  }, ((_d = userConfig.lessLoader) == null ? void 0 : _d.lessOptions) || {}), {
    modifyVars: userConfig.theme || {}
  });
  return config;
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getBrowserlist
});
